plugins/arUnogPreservicaPlugin/modules/arUnogPreservicaPlugin/actions/digitalobjectsCreateAction.class.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class arUnogPreservicaPluginDigitalObjectsCreateAction extends QubitApiAction
+{
+    protected function post($request, $payload)
+    {
+        $slug = $payload->slug;
+        $uuid = $payload->uuid;
+        $scopeId = $payload->scopeId;
+
+        // Fail if either a slug or UUID hasn't been provided
+        if (empty($slug) || empty($uuid)) {
+            $message = $this->context->i18n->__('Both slug and UUID must be specified.');
+
+            throw new QubitApiBadRequestException($message);
+        }
+
+        // Attempt to load information object
+        if (empty($io = QubitInformationObject::getBySlug($slug)) || 'QubitInformationObject' != get_class($io)) {
+            $message = $this->context->i18n->__('Information object not found.');
+
+            throw new QubitApiBadRequestException($message);
+        }
+
+        // Fail if a digital object already exists for the information object
+        if (!empty($io->getDigitalObject())) {
+            $message = $this->context->i18n->__('Information object already has a digital object.');
+
+            throw new QubitApiBadRequestException($message);
+        }
+
+        // Get Preservica digital object details
+        $username = sfConfig::get('app_preservica_username');
+        $password = sfConfig::get('app_preservica_password');
+        $host = sfConfig::get('app_preservica_host');
+
+        $client = new arUnogPreservicaPluginRestClient($host, $username, $password);
+
+        // Fail if credentials are invalid
+        if (empty($client->token)) {
+            $message = $this->context->i18n->__('Invalid token/credentials.');
+
+            throw new QubitApiUnknownException($message);
+        }
+
+        // Fail if no object with that UUID exists
+        if (empty($objectData = $client->getObjectDetails($uuid))) {
+            $message = $this->context->i18n->__('Error fetching object data.');
```

Comment:
For debugging purposes, it might be useful to have this explicitly say "Error fetching object data from Preservica"?